app/main.py

```diff
@@ -1,8 +1,33 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list[dict]) -> list:
+    person_list = [
+        Person.people.get(
+            person.get("name"),
+            Person(person.get("name"), person.get("age"))
+        ) for person in people
+    ]
+
+    for person in people:
+        wife_or_husband = "wife" if "wife" in person else "husband"
+        name_of_wife_or_husband = person[wife_or_husband]
```

Comment:
Not true
Maybe key not exist in total?

Reply:
Simply this code. It is too hard
No need to check wife_or_husband
You have only two options:
1. If wife exist 
2. If husband exist